src/server/pps/server/api_server.go

```diff
@@ -1130,6 +1140,17 @@ func (a *apiServer) scaleUpWorkers(ctx context.Context, deploymentName string, p
 	return err
 }
 
+func (a *apiServer) workerServiceIP(ctx context.Context, deploymentName string) (string, error) {
```

Comment:
Yes, that requires DNS, which I don't think we can count on all clusters having. We normally use env vars but that won't work here because they can't get injected after the pod is created.